scripts/benchmark/batching/results_comparison.py

```diff
@@ -0,0 +1,195 @@
+
+        metric = st.selectbox(
+            'Metric:',
+            options=[SELECTBOX_DEFAULT_VALUE] + BenchmarkResultColumn.metrics(),
+            help='Benchmark metric by which the comparison will be made.',
+        )
+
+    with right_column:
+        sample_size = st.number_input(
+            'Sample size:',
+            min_value=0,
+            max_value=dataset_size,
+            step=1,
+            help=(
+                'Number of projects in a sample. '
+                'If you did not use a sample, then set the value equal to the dataset size.'
+            ),
+        )
+
+        aggregation_function = AggregationFunction(
+            st.selectbox(
+                'Aggregation function:',
+                options=AggregationFunction.values(),
+                index=AggregationFunction.values().index(AggregationFunction.MEAN.value),
+                help='Aggregation function that will be used when grouping results.',
+            ),
+        )
+
+    if results_dir and dataset_size != 0 and sample_size != 0 and metric != SELECTBOX_DEFAULT_VALUE:
+        results = {
+            result_path.stem: pd.read_csv(result_path)
+            for result_path in get_all_file_system_items(results_dir, with_subdirs=False)
+        }
+
+        st.header('Comparison table')
+
+        comparison_table = compare_results(
+            results,
+            BenchmarkResultColumn(metric),
+            aggregation_function,
+            dataset_size,
+            sample_size,
+        )
+        st.write(comparison_table.style.highlight_min().set_precision(2))
+
+        with st.expander('Table description:'):
+            st.markdown(
+                r"""
+                There are 3 types of stats:
+                - `total` — Sum of the metric for all the batches (the metric are pre-aggregated within each batch).
+                - `per_project` — Approximate time it will take to analyze one project. Calculated by the formula:
+                  $$\texttt{total} \div \texttt{Sample size}$$.
+                - `dataset` — Approximate time it will take to analyze the entire dataset. Calculated by the formula:
+                  $$\texttt{per\_project} \cdot \texttt{Sample size}$$.
+
+                  If the dataset size is equal to the sample size, this statistic is omitted.
+
+                If the results contain several run types, the stats for the specific run type will be shown in addition
+                to the overall stats.
+
+                The minimum values of each statistic are highlighted in yellow.
+                """,
+            )
+
+        st.header('Per batch stats')
+        result_name = st.selectbox('Result name:', options=results.keys())
+        st.plotly_chart(
+            plot_per_batch_stats(results[result_name], BenchmarkResultColumn(metric), aggregation_function),
+            use_container_width=True,
+        )
+
+
+if __name__ == '__main__':
```

Comment:
Can we add simple tests that just run the main function and check if no errors occur?